### DISCUSSION TITLE
Code example on how to ConfirmUpdate?

### DISCUSSION THREAD

**Discussion by ainsleybernard** [MarkMpn/Sql4Cds#161] [Q&A] [answered]
Hi would it be possible to provide example on how to use      `   bool ConfirmUpdate(int count, EntityMetadata meta);`

As Im not sure where The parameter `entityMetadata` should come from.

Thanks!

**Comment by MarkMpn** [comment, marked as answer]
This method will be called by the engine just before records are about to be updated. The `count` parameter will be set to the number of records about to be updated, and the `meta` parameter will be set to the metadata of the entity type of those records.

If the method returns `true` the update continues, or it will be cancelled if it returns `false`.

You can implement this as a simple `return true;` to allow all updates to proceed, as @miseeger has done at https://github.com/miseeger/Dataverse.Sql/blob/f7b3800f72abad5ced69ccb2dc4d16c0695adcd8/Dataverse.Sql/QueryExecutionOptions.cs#L201, or you could present a prompt to the user to confirm whether the update should proceed as the SQL 4 CDS XrmToolBox tool does at https://github.com/MarkMpn/Sql4Cds/blob/1e2f5da4ea69de8f7fe479302204ff74b5ef9c55/MarkMpn.Sql4Cds/QueryExecutionOptions.cs#L73

**Reply by ainsleybernard** [reply]
Got it, thanks!
